mobilecoind-dev-faucet/src/worker.rs

```diff
@@ -18,24 +18,23 @@
 //! The worker does not require to be launched from the context of a tokio
 //! runtime.
 
+use displaydoc::Display;
 use mc_common::logger::{log, o, Logger};
 use mc_mobilecoind_api::{
     external::PublicAddress, mobilecoind_api_grpc::MobilecoindApiClient, SubmitTxResponse,
     TxStatus, UnspentTxOut,
 };
 use mc_transaction_core::{constants::MAX_OUTPUTS, ring_signature::KeyImage, TokenId};
 use std::{
+    cmp::max,
     collections::{hash_map::Entry, HashMap, HashSet},
     sync::{
         atomic::{AtomicBool, AtomicUsize, Ordering},
-        Arc, Mutex,
+        Arc,
     },
     time::Duration,
 };
-use tokio::sync::{
-    mpsc::{self, UnboundedReceiver, UnboundedSender},
-    oneshot::{self, error::TryRecvError},
-};
+use tokio::sync::oneshot::{self, error::TryRecvError};
```

Comment:
My feeling is that, using the one-shot sender when you only actually have to send one value is helpful:

* Aids local reasoning about the program. If I have only read a small part of the program and I see oneshot sender / receiver, I know that it can only be used once and is only supposed to be used once
* There are small differences in the API and the behavior
* Sending to a one-shot sender consumes Self which prevents you statically from trying to send again
* Sending to a one-shot sender cannot get blocked if the channel is already full, because it can't be
* Awaiting the receiver finishes exactly when an item is received, and there's no possibility that you sit around awaiting a second item.

The main reason I changed away from mpsc channel was that I actually do need a queue, and I don't just have a single consumer, and the tokio docs recommended async_channel for that, it seemed better than sticking a mutex around the receiver.

Reply:
Got it, thanks for clarifying.